core/capabilities/ccip/ccipevm/addresscodec.go

```diff
@@ -4,13 +4,14 @@ import (
 	"encoding/hex"
 	"fmt"
 	"strings"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
 )
 
 type AddressCodec struct{}
 
 func (a AddressCodec) AddressBytesToString(addr []byte) (string, error) {
-	// TODO support EIP-55 checksum, https://smartcontract-it.atlassian.net/browse/CCIP-5340
-	return "0x" + hex.EncodeToString(addr), nil
+	return gethcommon.BytesToAddress(addr).Hex(), nil
```

Comment:
Note: this will require an update on the EVM staging cluster

Reply:
If we are going to support EIP-55, maybe we should update the `AddressStringToBytes()` to use geth common library as well.

Reply:
Also adding a length check for the address, since `BytesToAddress` doesn't check it.

Reply:
check [this](https://github.com/smartcontractkit/chainlink/commit/82fade97ba57a8242abcf0c7d828f6137b17dd76#diff-62c6e868fa1f049d352664bd255091657013ada17214f1bd807f57f848547b15) as an example